### ISSUE TITLE
x13 and timetravel support 

### ISSUE THREAD

**Issue by solutionjet** [zone117x/node-stratum-pool#131]
hi, i am new to pool and nomp configuration and stuff. i somehow managed to configure and run it for ltc. I wonder how can i add support for coins that didn't ship with the bundle like marucoin/x13 and timetravel 10 algorithm?

thanks for any help..

**Comment by hardx4** [comment]
I was able to implement the Bitcore currency that works in timetravel10, but I can not solve the problem of rejected shares. 100% of the shares are rejected.

**Comment by kiahmed** [comment]
@hardx4 could you shed some more light on where to get a the algorithm files for each coin that this stratum pool would understand and work with? For bitcore, did you have to download timetravel10 file from anywhere or created your own or something?

**Comment by hardx4** [comment]
@kiahmed See my repository. Well, I downloaded the timetravel of a pool that supports the coin in your stratum. I downloaded it here: https://github.com/tpruvot/yiimp. It seems to work fine, but it does not accept the shares.

My issue: https://github.com/zone117x/node-stratum-pool/issues/132

**Comment by kiahmed** [comment]
yup. i looked into that repo too..he has wide range of coins setup and entire package. So, why wouldn't you rather use that stratum service see if it doesn't has the sharing problem or you already have infrastructure running on this repo? Thats exactly my concern is how do you make sure the stratum is all happy with the newly incorporated algo. I think I tried marucoin with this one and it had some issues.

**Comment by startsts** [comment]
I recently added this algo to my pool (I use node-stratum), mining seems working,  mulitplier at "algoProperties" I set 8.

**Comment by hardx4** [comment]
@startsts, Timetravel10 or Timetravel?

**Comment by hardx4** [comment]
My something is factor equal to 1, I'm going to try to change it, it might solve my problems.

**Comment by startsts** [comment]
In my case is Timetravel,  but I think Timetravel10 has same multiplier.   
Btw: do you have polytimos working at your multi-hashing?

**Comment by hardx4** [comment]
@startsts yes i have.

**Comment by hardx4** [comment]
@startsts I will try to change the timetravel10 multiplier and I will test, I believe it will work!

**Comment by startsts** [comment]
I have seen your code for timetravel10  https://github.com/hardx4/node-multi-hashing/blob/master/timetravel10.c      and wonder  it has only 7 algos instead of 10, from there is that code?

**Comment by hardx4** [comment]
This version is one of many that I tested, I will return to a version with 9 cases instead of 7

**Comment by startsts** [comment]
And yes... I was working on timetravel10 and got same problem with reject shares and it was not depend on multiplier.   I have taken some code from cpuminer-multi, now it seems working,   I pulled to repository, I hope this help  https://github.com/startsts/multi-hashing/blob/master/bitcore.c

**Comment by hardx4** [comment]
@startsts I tested the code but continued to reject the shares. So I put in multiplier: Math.pow (2, 8) and the shares were accepted. But I still do not know if this is correct.

**Comment by startsts** [comment]
@hardx4 yes, i was talking about Math.pow (2, 8)  above.  To know if wallet will accept block we can try to use testnet or some coin with low difficulty and try mining it.  I never created testnet so I don't know how to do it.     I recently added POLY coin to my pool,  but there are no miners yet, so nobody checked .  Any other coin with timetravel10?

**Comment by hardx4** [comment]
@startsts I do not know the same way. I tried to start in testnet, but apparently there are no nodes available. What is your pool?

**Comment by startsts** [comment]
my pool is https://cryptohub.online/pools/,  any ideas?     Surely there are no nodes for testnet, you should launch 2 wallet instances on the same computer and connect one to another

**Comment by hardx4** [comment]
I have bitcore installed on the server, I can start in testnet with your added node and you in the same way with my added node. Would that solve it?
How can we communicate more easily? skype, slack, discord ...

**Comment by startsts** [comment]
I don't have broadband connection these days so I can't help.  How I say you can launch 2 daemon instances in one PC,  just use different -datadir.   I used this way when was checking tribus algo

**Comment by hardx4** [comment]
@startsts I have just tested in testnet and the daemon refuses the generated blocks ...

**Comment by startsts** [comment]
Can you see in debug.log what the reason?

**Comment by startsts** [comment]
I tried testnet and everything is ok, daemon accepts blocks, screenshot
![screenshot at 2017-12-13 01-54-13](https://user-images.githubusercontent.com/5901966/33922301-a2c2c02c-dfa8-11e7-99e6-6a915d6d643a.png)

**Comment by hardx4** [comment]
@startsts  Here the blocks are not accepted =/

**Comment by hardx4** [comment]
@startsts  How's your shareMultiplier for the algorithm?
I'll give you the log, one minute

**Comment by startsts** [comment]
bitcore: {
        multiplier: Math.pow(2, 8),
        hash: function(){
            return function(){
                return multiHashing.bitcore.apply(this, arguments);
            }
        }
    },

**Comment by hardx4** [comment]
Let me compile again

**Comment by hardx4** [comment]
@startsts look
![image](https://user-images.githubusercontent.com/7688208/33922762-5b464a34-dfa3-11e7-8f8b-19fe2cd72cb2.png)

**Comment by hardx4** [comment]
@startsts If Daemon rejects the block, why does it add the block to the chain, this I do not understand ...

**Comment by startsts** [comment]
daemon accepts block, but your stratum thinks that it doesn't since it return different block hash ..  maybe need some changes to node-stratum-pool,   I already did a lot of changes in code of node-stratum-pool to support more coins... when I will be able I publish my node-stratum-pool repository

**Comment by hardx4** [comment]
@startsts I understood, I still do not know how to solve this problem.

**Comment by trustfarm-dev** [comment]
@hardx4  @startsts  can you check my problem also?
@hardx4  in case of me, making groestlmyriad algo pool. coin has different header hashing method.
in case of digibyte , they use blockhash with SHA256D , verge is use scrypt.
Can you check BTX source code what header hashing algorithm used.

**Comment by globaltoken** [comment]
@hardx4 I will copy your Code into my Repo. I am working to add dozens of algorithms for NOMP. About 25 - 30. What is the problem you have?

Regards,

**Comment by globaltoken** [comment]
@hardx4 I think I found your issue.
Seems you are hashing in wrong way mate.

You are hashing hashA, hashB, hashA, hashB. But it works this way: 
hashA, hashB, hashB, hashA, hashA, hashB ..

I will fix this and let you know more! @kiahmed @trustfarm-dev @startsts
